WordPress/Classes/Services/PostService.m

```diff
@@ -333,70 +337,137 @@ - (void)autoSave:(AbstractPost *)post
+        failureBlock(defaultError);
     }
 }
 
+- (void)handleAutoSaveWithRestRemote:(PostServiceRemoteREST *)restRemote
+                             forPost:(AbstractPost *)post
+                             success:(void (^)(AbstractPost *post, NSString *previewURL))success
+                             failure:(void (^)(NSError * _Nullable error))failure
+{
+    NSManagedObjectID *postObjectID = post.objectID;
+    RemotePost *remotePost = [self remotePostWithPost:post];
+
+    AutosaveSuccessBlock autosaveSuccessBlock = [self wrappedAutosaveSuccessBlock:postObjectID success:success];
+    AutosaveFailureBlock setPostAsFailedAndCallFailureBlock = [self wrappedAutosaveFailureBlock:post failure:failure];
+
+    // The autoSave endpoint returns an exception on posts that do not exist on the server
+    // so we'll create the post instead if necessary.
+    BOOL mustBeCreated = ![post hasRemote];
+
+    if (mustBeCreated) {
+        // Abort if the status is trashed/deleted. We'd rather not automatically create a
+        // locally trashed post as drafts in the server.
+        if ([post.status isEqualToString:PostStatusTrash] || [post.status isEqualToString:PostStatusDeleted]) {
+            NSError *error = [self defaultAutosaveError];
+            setPostAsFailedAndCallFailureBlock(error);
+            return;
+        }
+
+        [self uploadPost:post
+    forceDraftIfCreating:YES
+                 success:^(AbstractPost * _Nonnull post) {
+                     success(post, nil);
+                 }
+                 failure:failure];
+        return;
+    }
+
+    if ([post originalIsDraft]) {
+        // Calling the v1.1 autosave endpoint for a draft post will close
+        // the comments for the post. See https://github.com/wordpress-mobile/WordPress-iOS/issues/13079
+        // and p3hLNG-15Z-p2 for more background on this issue.
+        // For drafts, we can just call uploadPost and a new revision is
+        // automatically created anyway. We check original is draft since
+        // the post should be a draft on the server if we're uploading vs
+        // autosaving.
+        [restRemote getPostWithID:remotePost.postID success:^(RemotePost *tempPost) {
+            if ([tempPost.status isEqualToString:PostStatusDraft]) {
+
+                // We have to be careful about uploading when the status has been
+                // changed locally. Since the post is a draft on the server,
+                // use the draft status now, and restore to whatever the status
+                // might have been set to after the call completes.
+                NSString *savedStatus = post.status;
+                post.status = PostStatusDraft;
+
+                [self uploadPost:post
+                success:^(AbstractPost * _Nonnull post) {
+                    post.status = savedStatus;
+                    success(post, nil);
+                } failure:failure];
```

Comment:
Oh, I think we should also restore the `savedStatus` when the `failure` block is called.

Reply:
Doh!  Right you are.